example-app/santoku-app-backend/src/main/java/jp/fintan/mobile/santokuapp/system/nablarch/AccountInitializer.java

```diff
@@ -0,0 +1,103 @@
+package jp.fintan.mobile.santokuapp.system.nablarch;
+
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.SQLException;
+import java.util.List;
+import javax.sql.DataSource;
+import jp.fintan.mobile.santokuapp.domain.model.account.Account;
+import jp.fintan.mobile.santokuapp.domain.model.account.AccountId;
+import jp.fintan.mobile.santokuapp.domain.model.account.Nickname;
+import jp.fintan.mobile.santokuapp.domain.model.account.RawPassword;
+import jp.fintan.mobile.santokuapp.infrastructure.service.PBKDF2PasswordHashingProcessor;
+import nablarch.core.log.Logger;
+import nablarch.core.log.LoggerManager;
+import nablarch.core.repository.initialization.Initializable;
+import nablarch.core.util.StringUtil;
+
+/**
+ * 以下のアカウントを初期データとして登録します。
+ * <ul>
+ *   <li>santoku</li>
+ *   <li>admin</li>
+ *   <li>partner</li>
+ * </ul>
+ * <p>
+ * 既にアカウントが登録されている場合は、初期データを更新します。
+ * <p>
+ * パスワードは、環境設定ファイル（env.config）、または環境変数で設定可能です。
+ * <ul>
+ *   <li>環境設定ファイル: 「account.password」に設定</li>
+ *   <li>環境変数: 「ACCOUNT_PASSWORD」に設定</li>
+ * </ul>
+ *
+ * @see <a href="https://nablarch.github.io/docs/5u20/doc/application_framework/application_framework/setting_guide/ManagingEnvironmentalConfiguration/index.html">処理方式、環境に依存する設定の管理方法</a>
+ * @see <a href="https://nablarch.github.io/docs/5u20/doc/application_framework/application_framework/libraries/repository.html#os">OS環境変数を使って環境依存値を上書きする</a>
+ */
+public class AccountInitializer implements Initializable {
+
+  private static final Logger LOGGER = LoggerManager.get(AccountInitializer.class);
+  private DataSource dataSource;
+  public DataSource getDataSource() {
+    return dataSource;
+  }
+
+  public void setDataSource(DataSource dataSource) {
+    this.dataSource = dataSource;
+  }
+
+  private String accountPassword;
+  public String getAccountPassword() {
+    return accountPassword;
+  }
+
+  public void setAccountPassword(String accountPassword) {
+    this.accountPassword = accountPassword;
+  }
+  private static final Account SANTOKU_ACCOUNT = new Account(new AccountId("santoku"), new Nickname("santoku"));
+  private static final Account ADMIN_ACCOUNT = new Account(new AccountId("admin"), new Nickname("santoku.administrator.techlead.1234567890123456789"));
+  private static final Account PARTNER_ACCOUNT = new Account(new AccountId("partner"), new Nickname("p"));
+  private static final List<Account> ACCOUNTS = List.of(SANTOKU_ACCOUNT, ADMIN_ACCOUNT, PARTNER_ACCOUNT);
+
+  @Override
+  public void initialize() {
+    if (StringUtil.isNullOrEmpty(accountPassword)) {
+      LOGGER.logWarn("Initial account data could not be registered because no password was set in environment variable or system properties.");
+      return;
+    }
+    String hashedPassword = new PBKDF2PasswordHashingProcessor().hash(new RawPassword(accountPassword)).value();
```

Comment:
```java
    if (StringUtil.isNullOrEmpty(accountPassword)) {
      LOGGER.logWarn("Initial account data could not be registered because no password was set in environment variable or system properties.");
      return;
```

[RawPassword​.java](https://github.com/ws-4020/mobile-app-crib-notes/blob/2bd91697242c419ebc522a58ebcac817616851c0/example-app/santoku-app-backend/src/main/java/jp/fintan/mobile/santokuapp/domain/model/account/RawPassword.java#L11) 側で色々チェックしているので、まとめてログ出したほうが良いかも知れません

```java
  public RawPassword(String value) {
    super(value);
    validateNotBlank(value);
    validateLessThanOrEqual(value, MAX_LENGTH);
    validateGreaterThanOrEqual(value, MIN_LENGTH);
  }
```

- 未確認: `validateNotBlank` に null 渡した場合どうなるか？（そもそも null は来るのか？）

Reply:
バリデーションは`RawPassword`におまかせしました

> 未確認: validateNotBlank に null 渡した場合どうなるか？（そもそも null は来るのか？）

`env.config`に`account.password`がないと、↓のDIエラーが発生するのと、環境変数で`ACCOUNT_PASSWORD=`と設定した場合はブランクが入ってくるので、`null`の考慮は不要そうです。

```console
nablarch.core.repository.di.ConfigurationLoadException: property value was not found. parameter = ${account.password}
```

de01287a0c10e25312f5927efb40d78280dc187d